src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -647,7 +646,7 @@ ImmutableArray<byte> computeContentHash()
                     // Switch this to ImmutableCollectionsMarshal.AsImmutableArray(hash.GetHashAndReset()) when we move to S.C.I v8.
                     Span<byte> destination = stackalloc byte[128 / 8];
```

Comment:
nit, stack allocing here seems pointless, since we know we're going to an array at the end.  better to just new the array, write into the span for it, then do `ImmutableCollectionsMarshal.AsImmutableArray.

Now there will be zero coping of any sort.

Reply:
It would work by allocating and copying my point is that we should avoid that entirely by going straight to the array :-)